Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -1977,28 +1977,6 @@ WebPage& WebPage::fromCorePage(Page& page)
     return static_cast<WebChromeClient&>(page.chrome().client()).page();
 }
 
-static std::optional<FrameTreeNodeData> frameTreeNodeData(AbstractFrame& frame)
-{
-    std::optional<FrameTreeNodeData> info;
-    if (auto* webFrame = WebFrame::fromCoreFrame(frame)) {
-        Vector<FrameTreeNodeData> children;
-        for (auto* childFrame = frame.tree().firstChild(); childFrame; childFrame = childFrame->tree().nextSibling()) {
```

Comment:
So we used to get frame info data in tree order. However, after your change, I don't believe this is the case anymore. Although you switched to a ListHashSet to get some sort of a deterministic order, this is not the more logical tree order, which is very unfortunate.

I think the first step should be to have a FrameTree representation in the UIProcess (which we will need for process isolation no matter what). Then we could maintain tree order while re-implementing this.

Reply:
FrameTree::appendChild is the only way to add to a FrameTree right now, so creation order is the same order as what we had before.